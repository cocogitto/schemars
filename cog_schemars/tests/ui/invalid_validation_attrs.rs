use cog_schemars::JsonSchema;

#[derive(JsonSchema)]
#[validate(email)]
pub struct Struct1(#[validate(regex, foo, length(min = 1, equal = 2, bar))] String);

#[derive(JsonSchema)]
#[cog_schemars(email)]
pub struct Struct2(#[cog_schemars(regex, foo, length(min = 1, equal = 2, bar))] String);

#[derive(JsonSchema)]
pub struct Struct3(
    #[validate(
        regex = "foo",
        contains = "bar",
        regex(pattern = "baz"),
        regex(path = "baz"),
        phone,
        email,
        url
    )]
    String,
);

#[derive(JsonSchema)]
pub struct Struct4(
    #[cog_schemars(
        regex = "foo",
        contains = "bar",
        regex(path = "baz"),
        regex(pattern = "baz"),
        phone,
        email(code = "code_str", message = "message"),
        email = "foo",
        email,
        email,
        url
    )]
    String,
);

fn main() {}
