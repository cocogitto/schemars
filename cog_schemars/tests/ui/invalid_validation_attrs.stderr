error: expected validate regex attribute item to be of the form `regex(...)`
 --> tests/ui/invalid_validation_attrs.rs:5:31
  |
5 | pub struct Struct1(#[validate(regex, foo, length(min = 1, equal = 2, bar))] String);
  |                               ^^^^^

error: expected cog_schemars regex attribute item to be of the form `regex(...)`
 --> tests/ui/invalid_validation_attrs.rs:9:35
  |
9 | pub struct Struct2(#[cog_schemars(regex, foo, length(min = 1, equal = 2, bar))] String);
  |                                   ^^^^^

error: cog_schemars attribute cannot contain both `equal` and `min`
 --> tests/ui/invalid_validation_attrs.rs:9:63
  |
9 | pub struct Struct2(#[cog_schemars(regex, foo, length(min = 1, equal = 2, bar))] String);
  |                                                               ^^^^^^^^^

error: unknown item in cog_schemars length attribute: `bar`
 --> tests/ui/invalid_validation_attrs.rs:9:74
  |
9 | pub struct Struct2(#[cog_schemars(regex, foo, length(min = 1, equal = 2, bar))] String);
  |                                                                          ^^^

error: unknown cog_schemars attribute `foo`
 --> tests/ui/invalid_validation_attrs.rs:9:42
  |
9 | pub struct Struct2(#[cog_schemars(regex, foo, length(min = 1, equal = 2, bar))] String);
  |                                          ^^^

error: unknown cog_schemars attribute `email`
 --> tests/ui/invalid_validation_attrs.rs:8:16
  |
8 | #[cog_schemars(email)]
  |                ^^^^^

error: expected validate regex attribute item to be of the form `regex(...)`
  --> tests/ui/invalid_validation_attrs.rs:14:9
   |
14 |         regex = "foo",
   |         ^^^^^^^^^^^^^

error: expected validate contains attribute item to be of the form `contains(...)`
  --> tests/ui/invalid_validation_attrs.rs:15:9
   |
15 |         contains = "bar",
   |         ^^^^^^^^^^^^^^^^

error: `pattern` is not supported in `validate(regex(...))` attribute - use either `validate(regex(path = ...))` or `cog_schemars(regex(pattern = ...))` instead
  --> tests/ui/invalid_validation_attrs.rs:16:15
   |
16 |         regex(pattern = "baz"),
   |               ^^^^^^^^^^^^^^^

error: `validate(regex(...))` attribute requires `path = ...`
  --> tests/ui/invalid_validation_attrs.rs:16:9
   |
16 |         regex(pattern = "baz"),
   |         ^^^^^^^^^^^^^^^^^^^^^^

error: expected cog_schemars regex attribute item to be of the form `regex(...)`
  --> tests/ui/invalid_validation_attrs.rs:28:9
   |
28 |         regex = "foo",
   |         ^^^^^^^^^^^^^

error: expected cog_schemars contains attribute item to be of the form `contains(...)`
  --> tests/ui/invalid_validation_attrs.rs:29:9
   |
29 |         contains = "bar",
   |         ^^^^^^^^^^^^^^^^

error: `path` is not supported in `cog_schemars(regex(...))` attribute - use `cog_schemars(regex(pattern = ...))` instead
  --> tests/ui/invalid_validation_attrs.rs:30:15
   |
30 |         regex(path = "baz"),
   |               ^^^^^^^^^^^^

error: `cog_schemars(regex(...))` attribute requires `pattern = ...`
  --> tests/ui/invalid_validation_attrs.rs:30:9
   |
30 |         regex(path = "baz"),
   |         ^^^^^^^^^^^^^^^^^^^

error: unexpected value of cog_schemars email attribute item
  --> tests/ui/invalid_validation_attrs.rs:33:14
   |
33 |         email(code = "code_str", message = "message"),
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: unexpected value of cog_schemars email attribute item
  --> tests/ui/invalid_validation_attrs.rs:34:15
   |
34 |         email = "foo",
   |               ^^^^^^^

error: duplicate cog_schemars attribute item `email`
  --> tests/ui/invalid_validation_attrs.rs:36:9
   |
36 |         email,
   |         ^^^^^

error: cog_schemars attribute cannot contain both `url` and `email`
  --> tests/ui/invalid_validation_attrs.rs:37:9
   |
37 |         url
   |         ^^^

error: unknown cog_schemars attribute `phone`
  --> tests/ui/invalid_validation_attrs.rs:32:9
   |
32 |         phone,
   |         ^^^^^
